File: python/sieve/sieve.py
def primes(limit):
    table = set(range(2, limit + 1))
    result = []
    while len(table):
        min_val = min(table)
        result.append(min_val)
        to_sub = set(range(min_val, limit + 1, min_val))
        table -= to_sub
    return result

File: javascript/sieve/sieve.js
export function primes(limit) {
  if (limit < 2) return [];
  let primeSet = new Set();
  for (let i = 2; i <= limit; ++i) {
    primeSet.add(i);
  }
  for (let i = 2; i <= limit; ++i) {
    if (primeSet.has(i)) {
      for (let j = 2; i * j <= limit; ++j)
        primeSet.delete(i * j);
    }
  }
  return Array.from(primeSet);
}
